src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/config/overrides/ConfigOverridesApplier.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.config.overrides;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.PerformanceAnalyzerApp;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.RcaConf;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.reader.ClusterDetailsEventProcessor;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import org.apache.commons.lang3.math.NumberUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Class the manages applying the various overrides for RCAs, deciders, and action nodes as
+ * written by the NodeDetails collector and read by the {@link ClusterDetailsEventProcessor}
+ */
+public class ConfigOverridesApplier {
+
+  private static final Logger LOG = LogManager.getLogger(ConfigOverridesApplier.class);
+  private volatile long lastAppliedTimestamp;
+
+  public void applyOverride(final String overridesJson, final String lastUpdatedTimestampString) {
+    if (!valid(overridesJson, lastUpdatedTimestampString)) {
+      LOG.warn("Received invalid overrides or timestamp. Overrides json: {}, last updated "
+          + "timestamp: {}", overridesJson, lastAppliedTimestamp);
+      return;
+    }
+
+    try {
+      long lastUpdatedTimestamp = Long.parseLong(lastUpdatedTimestampString);
+      LOG.debug("Last updated(writer): {}, Last applied(reader): {}", lastUpdatedTimestamp,
+          lastAppliedTimestamp);
+      if (lastUpdatedTimestamp > lastAppliedTimestamp) {
+        apply(ConfigOverridesHelper.deserialize(overridesJson));
+      } else {
+        LOG.debug("Not applying override. Last updated timestamp {} is behind last applied "
+            + "timestamp {}", lastUpdatedTimestamp, lastAppliedTimestamp);
+      }
+    } catch (IOException ioe) {
+      LOG.error("Unable to deserialize overrides JSON:" + overridesJson, ioe);
+    }
+  }
+
+  private void apply(final ConfigOverrides overrides) {
+    if (PerformanceAnalyzerApp.getRcaController() != null && PerformanceAnalyzerApp.getRcaController().isRcaEnabled()) {
+      LOG.info("Applying overrides: {}", overrides.getEnable().getRcas());
+      RcaConf rcaConf = PerformanceAnalyzerApp.getRcaController().getRcaConf();
```

Comment:
Should we get the list of rca.conf from the RcaControllerHelper and apply to all three of them ? If the node role changes, then the muting will be invalidated.

Any changes that you make to the static set of objects in PerformanceAnalyzerApp may not be tested using RCA-IT.

Reply:
Good catch! updating all conf files now.